torchmetrics/detection/map.py

```diff
@@ -343,6 +372,9 @@ def _get_classes(self) -> List:
         return []
 
     def _compute_iou(self, id: int, class_id: int, max_det: int) -> Tensor:
+        return self._compute_iou_impl(id, class_id, max_det, segm_iou if self.iou_type == "segm" else box_iou)
```

Comment:
since you habe both states for metrics and masks, what if people want to calculate iou for both?